delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchDefaultViewsResetter.scala

```diff
@@ -0,0 +1,104 @@
+}
+
+object ElasticSearchDefaultViewsResetter {
+
+  private val logger = Logger[ElasticSearchDefaultViewsResetter]
+
+  def resetTrigger: IO[Boolean] =
+    Env[IO].get("RESET_DEFAULT_ES_VIEWS").map(_.getOrElse("false").toBoolean)
+
+  /**
+    * Provides a resetter that for each default elasticsearch view:
+    *
+    *   1. Deletes the associated index in elasticsearch 2. Deletes all associated events, states, and projection
+    *      offsets 3. Creates a new default elasticsearch view using the provided new view value
+    *
+    * Note that it will reset the default view even its project is deprecated.
+    */
+  def apply(
+      client: ElasticSearchClient,
+      views: ElasticSearchViews,
+      newViewValue: ElasticSearchViewValue,
+      xas: Transactors
+  )(implicit subject: Subject): ElasticSearchDefaultViewsResetter =
+    apply(
+      views.currentIndexingViews,
+      client.deleteIndex,
+      views.internalCreate(_, _, _)(subject).void,
+      newViewValue,
+      resetTrigger,
+      xas
+    )
+
+  def apply(
+      views: ElemStream[IndexingViewDef],
+      deleteIndex: IndexLabel => IO[Boolean],
+      createView: (Iri, ProjectRef, ElasticSearchViewValue) => IO[Unit],
+      newViewValue: ElasticSearchViewValue,
+      resetTrigger: IO[Boolean],
+      xas: Transactors
+  ): ElasticSearchDefaultViewsResetter =
+    new ElasticSearchDefaultViewsResetter {
+      override def resetDefaultViews: IO[Unit] =
+        resetTrigger.flatMap { triggered =>
+          IO.whenA(triggered) {
+            views
+              .filter(_.id == defaultEsViewId)
+              .flatTap { view => Stream.eval(view.evalMap(resetView)) }
+              .compile
+              .drain
+          }
+        }
+
+      override def resetView(view: IndexingViewDef): IO[Unit] =
+        deleteEsIndex(view) >>
+          deleteEventsStatesOffsets(view.ref.project).transact(xas.write) >>
+          createDefaultView(view.ref.project)
+
+      private val defaultEsViewId = iri"https://bluebrain.github.io/nexus/vocabulary/defaultElasticSearchIndex"
+
+      private def deleteEsIndex(viewDef: IndexingViewDef) =
+        viewDef match {
+          case activeView: ActiveViewDef => deleteIndex(activeView.index)
+          case _: DeprecatedViewDef      => IO.pure(true)
+        }
+
+      private def deleteEventsStatesOffsets(project: ProjectRef): doobie.ConnectionIO[Unit] =
+        sql"""
+          DELETE FROM scoped_events WHERE type = 'elasticsearch' AND id = ${defaultEsViewId.toString} AND org = ${project.organization} AND project = ${project.project};
+          DELETE FROM scoped_states WHERE type = 'elasticsearch' AND id = ${defaultEsViewId.toString} AND org = ${project.organization} AND project = ${project.project};
+          DELETE FROM projection_offsets WHERE module = 'elasticsearch' AND resource_id = ${defaultEsViewId.toString} AND project = $project;
+        """.stripMargin.update.run.void
```

Comment:
Given the fragile nature of the resetter tests, I prefer to hardcode the default view id in here so that we cannot delete something else by mistake. We should handle resource deletion properly in the future